### PR TITLE
Approve+request

### PR DESCRIPTION
**Pull request by laythq**

### CODE REVIEW

File: spec/features/rejectRequestSpec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+feature 'Reject Request' do
+  scenario 'User rejects a request' do
+    visit('/')
+    fill_in('username', :with => 'Daniel')
```
Comment (elishkaflint) [inline spec/features/rejectRequestSpec.rb]: Should we extract some of these activities into a web helper? Let's decide whether that's a priority at this point.

File: lib/listing.rb [hunk 2]
```diff
@@ -12,8 +14,18 @@ def initialize(id, name, owner_id, description, price)
 
   def self.create(name, owner_id, description, price)
     Listing.switch_database
+#     result = @connection.exec("INSERT INTO listings (name, owner_id) VALUES('#{name}', '#{owner_id}') RETURNING id, name, owner_id")
```
Comment (elishkaflint) [inline lib/listing.rb]: ok with this hashing but let's remove after we've checked that rspec still passes once we've merged (and conflict resolution wasn't an issue)

File: spec/request_spec.rb [hunk 3]
```diff
@@ -2,22 +2,64 @@
...
+    it 'returns a specific listing' do
+      user = User.add('username', 'password')
+      listing = Listing.create('Holiday_Home', user.id)
+      request = Request.return_request(listing.id)
+      expect(request.listing_id).to eq listing.id
+    end
+  end
+
+  describe '.reject' do
+    it 'rejects a request' do
```
Comment (elishkaflint) [inline spec/request_spec.rb]: Should we be thinking about isolating tests like this?

File: lib/request.rb [hunk 4]
```diff
@@ -1,28 +1,65 @@
...
+    Request.new(result.first['id'],result.first['listing_id'], result.first['requester_id'], result.first['host_id'], result.first['is_requested'], result.first['is_approved'])
+  end
+
+  def self.reject(request_id)
+    Request.switch_database
+    result = @connection.exec("UPDATE requests SET is_approved = FALSE, is_requested = FALSE WHERE id = #{request_id} RETURNING id, listing_id, requester_id, host_id, is_requested, is_approved;")
+    Request.new(result.first['id'],result.first['listing_id'], result.first['requester_id'], result.first['host_id'], result.first['is_requested'], result.first['is_approved'])
+  end
+
+  def self.available
```
Comment (elishkaflint) [inline lib/request.rb]: i don't fully understand what this is doing, and not sure that the corresponding test is testing the right thing?

File: app.rb [hunk 5]
```diff
@@ -1,6 +1,7 @@
 require 'sinatra/base'
```
Comment (elishkaflint) [inline app.rb]: BRILLIANT WORK LAYTH - the amount you've got done today is pretty phenomenal 😸

File: spec/features/rejectRequestSpec.rb (hunk 1 above)
Comment (laythq) [reply spec/features/rejectRequestSpec.rb]: Yeah we definitely need some quick methods for sign-in. Can work on it when we've done the MVP?

File: lib/listing.rb (hunk 2 above)
Comment (laythq) [reply lib/listing.rb]: Ok

File: app.rb (hunk 5 above)
Comment (laythq) [reply app.rb]: Thanks! :)

File: lib/request.rb (hunk 4 above)
Comment (laythq) [reply lib/request.rb]: It returns all available spaces, but I'll have a look at the test and make sure it's testing that.

**Pull request merged**